### PR TITLE
Fix Invalid type NoneType for attribute X error | AWS-Lambda instrumentation

### LINKED ISSUE #1780: opentelemetry-instrumentation-aws-lambda | Invalid type NoneType for attribute 'http.user_agent' value.
Environment:

I'm using `opentelemetry-instrumentation-aws-lambda` to instrument the Lambda function behind the API Gateway (V2).

Python==3.9
AWS-OTEL Lmabda Layer == arn:aws:lambda:us-east-1:901920570463:layer:aws-otel-python-amd64-ver-1-17-0:1
fastapi==0.73.0
opentelemetry-sdk==1.17.0

Environment variables:

```bash
AWS_LAMBDA_EXEC_WRAPPER=/opt/otel-instrument
OPENTELEMETRY_COLLECTOR_CONFIG_FILE=/var/task/src/otel-config.yaml
OTEL_METRICS_EXPORTER=otlp
OTEL_EXPORTER_OTLP_PROTOCOL=http/protobuf
OPENTELEMETRY_EXTENSION_LOG_LEVEL=error
OTEL_PYTHON_DISABLED_INSTRUMENTATIONS=fastapi
```

OTEL Config yaml:

```yaml
receivers:
  otlp:
    protocols:
      http:

exporters:
  otlp:
    endpoint: https://<ANOTHER_SERVICE>:4317

service:
  pipelines:
    metrics:
      receivers: [otlp]
      exporters: [otlp]

```

**Steps to reproduce**
1. Instrument your Lambda with as stated here: https://aws-otel.github.io/docs/getting-started/lambda/lambda-python
2. Lambda should be behind the API Gateway
3. Send a request

**What is the expected behavior?**
There shouldn't be any Warning logs in the Lambda logs.

**What is the actual behavior?**
I'm seeing couple of Warning in the lambda logs.

Screenshot 2023-05-03 at 9 30 27 AM

```bash
[WARNING]	2023-05-02T23:36:45.395Z	ec784472-c5bb-4f74-867d-e45d39e0fcca	Invalid type NoneType for attribute 'http.user_agent' value. Expected one of ['bool', 'str', 'bytes', 'int', 'float'] or a sequence of those types
```

**Additional context**
I'm interested to open a PR if possible to add some logic here preventing None values being set as attributes: https://github.com/open-telemetry/opentelemetry-python-contrib/blob/46e4b1da44c534fed8e1002899e9e41e6d668018/instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py#LL232C3-L232C3

Also, if there is a way to disable Tracing/Warning Logs  it would be helpful, I couldn't find a way to do that.

Another related discussion: https://github.com/open-telemetry/opentelemetry-python/discussions/3293

Thanks.

### PR DESCRIPTION
**Pull request by imanshafiei540**
# Description

This PR adds a `None` type checking to the `instrumentation-aws-lambda` logic. By checking attributes like `User-Agent` which can be `None`, this PR prevents adding `None` value attributes to the current `span`.

As mentioned in the issue page, this will prevent these errors (Especially for Lambda functions behind the Application Load Balancer):

```bash
[WARNING]	2023-05-02T23:36:45.395Z	ec784472-c5bb-4f74-867d-e45d39e0fcca	Invalid type NoneType for attribute 'http.user_agent' value. Expected one of ['bool', 'str', 'bytes', 'int', 'float'] or a sequence of those types
```

Fixes #1780 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] `tox -e py3{8,9,10,11}-test-instrumentation-aws-lambda`

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Review by srikanthccv — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py [hunk 1]
```diff
@@ -204,24 +204,30 @@ def _set_api_gateway_v1_proxy_attributes(
...
-            lambda_event["headers"].get("User-Agent"),
-        )
-        span.set_attribute(
-            SpanAttributes.HTTP_SCHEME,
-            lambda_event["headers"].get("X-Forwarded-Proto"),
-        )
-        span.set_attribute(
-            SpanAttributes.NET_HOST_NAME, lambda_event["headers"].get("Host")
-        )
+        if lambda_event["headers"].get("User-Agent"):
```
[the remark this answers is not in the document]
Comment (imanshafiei540) [reply instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py:207]: Thanks for the comment. Yes, that's correct. I thought that we didn't care about empty strings, but I was clearly wrong. I'll fix it in my next commit. Thanks.

[the remark this answers is not in the document]
Comment (imanshafiei540) [reply instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py]: Now, it's fixed. Thanks.

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,6 +39,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#1738](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1738))
 - Fix `None does not implement middleware` error when there are no middlewares registered
   ([#1766](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1766))
+- Fix `Invalid type NoneType for attribute X  (opentelemetry-instrumentation-aws-lambda)` error when some attributes do not exist
```
Comment (shalevr) [inline CHANGELOG.md:42]: Please move this entry to the Unreleased section

Comment (imanshafiei540) [reply CHANGELOG.md]: Updated. Thanks.

**Review by srikanthccv — APPROVED** [review]

**Review by shalevr — APPROVED** [review]

**Pull request merged**